shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/resetpassword/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package co.nimblehq.blisskmmic.presentation.modules.resetpassword
+
+import co.nimblehq.blisskmmic.MR
+import co.nimblehq.blisskmmic.data.network.helpers.toErrorMessage
+import co.nimblehq.blisskmmic.domain.usecase.ResetPasswordUseCase
+import co.nimblehq.blisskmmic.presentation.model.NotificationUiModel
+import co.nimblehq.blisskmmic.presentation.modules.BaseViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+data class ResetPasswordViewState(
+    val successNotification: NotificationUiModel? = null,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    constructor() : this(null, false, null)
```

Comment:
Do we really need this `constructor() : this(null, false, null)` as there is a default value to each [primary constructor](https://kotlinlang.org/docs/reference/classes.html#constructors) parameter already? 🤔

Reply:
If I don't include this, iOS cannot use the default constructor. 
<img width="779" alt="Screen Shot 2022-12-01 at 17 52 11" src="https://user-images.githubusercontent.com/6356137/205034601-2eeee2f8-f65f-4055-86a0-d4ea95998b5d.png">

Reply:
oh this is new knowledge to me, thanks for sharing 👍